code/datums/components/profound_fisher.dm

```diff
@@ -37,14 +40,26 @@
 	examine_list += span_info("When [EXAMINE_HINT("held")] or [EXAMINE_HINT("equipped")], [EXAMINE_HINT("right-click")] with a empty hand to open the integrated fishing rod interface.")
 	examine_list += span_tinynoticeital("To fish, you need to turn combat mode off.")
 
-/datum/component/profound_fisher/proc/on_rod_qdel(datum/source)
+///Handles replacing the fishing rod if somehow removed from the parent movable if delete_rod is TRUE, otherwise delete the component.
+/datum/component/profound_fisher/proc/on_rod_moved(datum/source)
 	SIGNAL_HANDLER
-	qdel(src)
+	if(QDELETED(src) || our_rod.loc == parent)
+		return
+	if(delete_rod)
+		UnregisterSignal(our_rod, COMSIG_MOVABLE_MOVED)
+		if(!QDELETED(our_rod))
+			qdel(our_rod)
+		our_rod = new our_rod.type(parent)
+	else
+		qdel(src)
 
 /datum/component/profound_fisher/Destroy()
-	our_rod.internal = FALSE
-	UnregisterSignal(our_rod, COMSIG_QDELETING)
-	REMOVE_TRAIT(our_rod, TRAIT_NOT_BARFABLE, REF(src))
+	UnregisterSignal(our_rod, COMSIG_MOVABLE_MOVED)
+	if(!delete_rod)
+		our_rod.internal = FALSE
+		REMOVE_TRAIT(our_rod, TRAIT_NOT_BARFABLE, REF(src))
+	else if(QDELETED(our_rod))
+		QDEL_NULL(our_rod)
```

Comment:
Not sure this is right?
If the rod is already being deleted you don't need to delete it again, just set it to null (which you will do anyway). If this code was instead _intended_ to delete the rod on component removal, I think it doesn't do that.

Reply:
Oh, good catch